### PR TITLE
LiveCodeBench leaderboard - release Tuesday

### PR DESCRIPTION
**Pull request by clefourrier**
Ready for a light review!

### CODE REVIEW

File: leaderboard-livecodebench.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+pip install -r requirements.txt
+```
+
+2. For evaluating new Hugging Face models, you can easily evaluate the model using 
+
+```
+python -m lcb_runner.runner.main --model {model_name} --scenario {scenario_name}
+```
+
+for different scenarios. For new model families, we have implemented an extensible framework and you can support new models by modifying `lcb_runner/lm_styles.py` and `lcb_runner/prompts` as described in the [github README](https://github.com/LiveCodeBench/LiveCodeBench).
```
Comment (clefourrier) [inline leaderboard-livecodebench.md:74]: Need a step `3. Once your results are generated, you can submit them doing ...`

File: leaderboard-livecodebench.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+## Findings
+
+We find that:
+- while model performances are correlated across different scenarios, the relative performances and orderings can vary on the 4 scenarios we use 
+- `GPT-4-Turbo` is the best-performing model across most scenarios but its margin grows on self-repair tasks highlighting its capability to take compiler feedback.
+- `Claude-3-Opus` overtakes `GPT-4-Turbo` in the test output prediction scenario highlighting stronger natural language reasoning capabilities. 
+- `Mistral-Large` performs considerably better on natural language reasoning tasks like test output prediction and code execution.
+
+![Performance on the 4 scenarios](https://github.com/LiveCodeBench/LiveCodeBench/raw/main/assets/images/tasks_radar.png)
+
```
Comment (clefourrier) [inline leaderboard-livecodebench.md:54]: Could use a small `## How to contribute` section, to tell the community how they can contact you, if they can give a hand, and link all the relevant information (your dataset, your own blog, the github, etc...)

File: leaderboard-livecodebench.md (hunk 1 above)
Comment (clefourrier) [reply leaderboard-livecodebench.md:74]: @Naman-ntc

File: leaderboard-livecodebench.md (hunk 2 above)
Comment (clefourrier) [reply leaderboard-livecodebench.md:54]: @Naman-ntc

File: _blog.yml [hunk 3]
```diff
@@ -3766,4 +3766,16 @@
...
     - guide
-    - trl
\ No newline at end of file
+    - trl
+
+- local: leaderboard-livecodebench
+  title: "Introducing the LiveCodeBench Leaderboard - Holistic and Contamination-Free Evaluation of Code LLMs"
+  author: StringChaos
+  guest: true
+  thumbnail: /blog/assets/leaderboards-on-the-hub/thumbnail.png
```
Comment (pcuenca) [inline _blog.yml:3775]: no custom thumbnail?

File: leaderboard-livecodebench.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+# Introducing the LiveCodeBench Leaderboard - Holistic and Contamination-Free Evaluation of Code LLMs
+
+We are excited to introduce the LiveCodeBench leaderboard, based on LiveCodeBench, a new benchmark developed by researchers from UC Berkeley, MIT, and Cornell for measuring LLMs’ code generation capabilities. 
+
+<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.45.1/gradio.js"> </script>
+<gradio-app theme_mode="light" space="livecodebench/leaderboard"></gradio-app>
+
+
+LiveCodeBench collects new problems from coding contest platforms over time annotated with problem release dates. It uses these annotations to evaluate models on problems released in different time windows allowing “evaluation over time” to detect and prevent contamination. Beyond code generation, LiveCodeBench assesses self-repair, test output prediction, and code execution, thus providing a more holistic view of coding capabilities required for the next generation of AI programming agents.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:28]:
```suggestion
LiveCodeBench collects coding problems over time from various coding contest platforms, annotating problems with their release dates. Annotations are used to evaluate models on problem sets released in different time windows, allowing an “evaluation over time” strategy that helps detect and prevent contamination. In addition to the usual code generation task, LiveCodeBench also assesses self-repair, test output prediction, and code execution, thus providing a more holistic view of coding capabilities required for the next generation of AI programming agents.
```

File: leaderboard-livecodebench.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+
+## LiveCodeBench Scenarios and Evaluation
+
+LiveCodeBench problems are curated from coding competition platforms: LeetCode, AtCoder, and CodeForces. These websites periodically host contests containing problems that assess the coding and problem-solving skills of participants. The problems consist of a natural language problem statement along with example input-output examples, and the goal is to write a program that passes a set of hidden tests. Further, thousands of participants participate, ensuring that the problems are vetted for clarity and correctness.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:33]:
```suggestion
LiveCodeBench problems are curated from coding competition platforms: LeetCode, AtCoder, and CodeForces. These websites periodically host contests containing problems that assess the coding and problem-solving skills of participants. Problems consist of a natural language problem statement along with example input-output examples, and the goal is to write a program that passes a set of hidden tests. Thousands of participants engage in the competitions, which ensures that the problems are vetted for clarity and correctness.
```

File: leaderboard-livecodebench.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+LiveCodeBench uses the collected problems for building its four coding scenarios
+
+- **Code Generation.** The model is given a problem statement, which includes a natural language description and example tests (input-output pairs) and is tasked with generating a correct solution. Evaluation is based on the functional correctness of the generated code, which is determined using a set of unseen test cases.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:37]:
```suggestion
- **Code Generation.** The model is given a problem statement, which includes a natural language description and example tests (input-output pairs), and is tasked with generating a correct solution. Evaluation is based on the functional correctness of the generated code, which is determined using a set of unseen test cases.
```

Comment (pcuenca) [reply leaderboard-livecodebench.md:37]: How are the test cases evaluated? Is the code sent to the respective platforms, did you write your own tests, or were they provided by the platforms (and kept hidden)?

File: leaderboard-livecodebench.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+- **Self Repair.** The model is given a problem statement and generates a candidate program, similar to the code generation scenario above. In case of a mistake, the model is provided with error feedback (either an exception message or a failing test case) and is tasked with generating a fixed solution. Evaluation is performed using functional correctness as defined above.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:38]:
```suggestion
- **Self Repair.** The model is given a problem statement and generates a candidate program, similar to the code generation scenario above. In case of a mistake, the model is provided with error feedback (either an exception message or a failing test case) and is tasked with generating a fix. Evaluation is performed using the same functional correctness as above.
```

File: leaderboard-livecodebench.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+- **Code Execution.** The model is provided a program snippet consisting of a function (f) along with a test input to the program and is tasked with predicting the output of the program on the input test case. Evaluation is based on an execution-based correctness metric: the model's output is considered correct if the assertion `assert f(input) == generated_output` passes.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:39]:
```suggestion
- **Code Execution.** The model is provided a program snippet consisting of a function (f) along with a test input, and is tasked with predicting the output of the program on the input test case. Evaluation is based on an execution-based correctness metric: the model's output is considered correct if the assertion `assert f(input) == generated_output` passes.
```

Is it a function or a program?

File: leaderboard-livecodebench.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+- **Test Output Prediction.** The model is given the problem statement along with a test case input and is tasked with generating the expected output for the input.  The tests are generated solely from problem statements, without the need for the function’s implementation and evaluated using an exact match checker.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:40]:
```suggestion
- **Test Output Prediction.** The model is given the problem statement along with a test case input and is tasked with generating the expected output for the input. Tests are generated solely from problem statements, without the need for the function’s implementation, and outputs are evaluated using an exact match checker.
```

When I first read the name of the task I thought it would be about predicting the results of unit tests. Maybe it could be called something like "Test Case Completion"? (Just a suggestion for the future, in case you think it's useful).

File: leaderboard-livecodebench.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+For each scenario, evaluation is performed using the Pass@1 metric. The metric captures the probability of generating a correct answer and is computed using the ratio of the count of correct answers over the count of total attempts, following `Pass@1 = total_correct / total_attempts`.
+
+## Preventing Benchmark Contamination
+
+Contamination is one of the major bottlenecks in current LLM evaluations. Even within LLM coding evaluations, there have been evidential reports of contamination and overfitting on standard benchmarks like HumanEval ([[1]](https://arxiv.org/abs/2403.05530) and [[2]](https://arxiv.org/abs/2311.04850)). 
+
+For this reason, we annotate problems with release dates in LiveCodeBench: that way, for new models with a training-cutoff date D, we can compute scores on problems released after D to measure their generalization on unseen problems. 
+
+LiveCodeBench formalizes this with a “scrolling over time” feature allowing naturally selecting problems in certain time windows, that you can try out in the leaderboard above!
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:53]:
```suggestion
LiveCodeBench formalizes this with a “scrolling over time” feature, that allows you to select problems within a specific time window. You can try it out in the leaderboard above!
```

File: leaderboard-livecodebench.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+## Findings
+
+We find that:
+- while model performances are correlated across different scenarios, the relative performances and orderings can vary on the 4 scenarios we use 
+- `GPT-4-Turbo` is the best-performing model across most scenarios but its margin grows on self-repair tasks highlighting its capability to take compiler feedback.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:59]:
```suggestion
- `GPT-4-Turbo` is the best-performing model across most scenarios. Furthermore, its margin grows on self-repair tasks, highlighting its capability to take compiler feedback.
```

File: leaderboard-livecodebench.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+- `Claude-3-Opus` overtakes `GPT-4-Turbo` in the test output prediction scenario highlighting stronger natural language reasoning capabilities.
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:60]:
```suggestion
- `Claude-3-Opus` overtakes `GPT-4-Turbo` in the test output prediction scenario, highlighting stronger natural language reasoning capabilities. 
```

File: leaderboard-livecodebench.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+python -m lcb_runner.runner.main --model {model_name} --scenario {scenario_name}
+```
+
+for different scenarios. For new model families, we have implemented an extensible framework and you can support new models by modifying `lcb_runner/lm_styles.py` and `lcb_runner/prompts` as described in the [github README](https://github.com/LiveCodeBench/LiveCodeBench).
+
+3. Once you results are generated, you can submit them by filling out [this form](https://forms.gle/h2abvAHh6UnhWzzd9).
+
+
+## How to contribute
+Finally, we are looking for collaborators and suggestions for LiveCodeBench. The [dataset](https://huggingface.co/livecodebench) and [code](https://github.com/LiveCodeBench/LiveCodeBench) are available on online and please reach out by submitting an issue or [mail](mailto:naman_jain@berkeley.edu).
```
Comment (pcuenca) [inline leaderboard-livecodebench.md:90]:
```suggestion
Finally, we are looking for collaborators and suggestions for LiveCodeBench. The [dataset](https://huggingface.co/livecodebench) and [code](https://github.com/LiveCodeBench/LiveCodeBench) are available online, so please reach out by submitting an issue or [mail](mailto:naman_jain@berkeley.edu).
```

**Review by pcuenca — APPROVED** [review]
Super interesting!

File: _blog.yml (hunk 3 above)
Comment (clefourrier) [reply _blog.yml:3775]: Nope not this time

File: leaderboard-livecodebench.md (hunk 9 above)
Comment (Naman-ntc) [reply leaderboard-livecodebench.md:40]: Oh thanks for the feedback!

File: leaderboard-livecodebench.md (hunk 6 above)
Comment (Naman-ntc) [reply leaderboard-livecodebench.md:37]: It varies across platforms. In some cases, test cases are released after the competition is over. In other cases, we collected the ground truth solutions, sampled inputs and get the expected output for the problems

File: leaderboard-livecodebench.md (hunk 8 above)
Comment (Naman-ntc) [reply leaderboard-livecodebench.md:39]: The entire program is a single function.

File: leaderboard-livecodebench.md (hunk 6 above)
Comment (clefourrier) [reply leaderboard-livecodebench.md:37]: Ok, removing unseen since it can sometimes contain publicly available data (it's not only your own private dataset)
